pkg/readiness/ready_tracker.go

```diff
@@ -95,6 +104,8 @@ func (t *Tracker) For(gvk schema.GroupVersionKind) Expectations {
 		return t.templates
 	case gvk.GroupVersion() == configv1alpha1.GroupVersion && gvk.Kind == "Config":
 		return t.config
+	case gvk.GroupVersion() == mutationv1alpha.GroupVersion && gvk.Kind == "AssignMetadata":
+		return t.assignMetadata
```

Comment:
This can be nil with the latest flag change - if that is the case it is safer to return a noopExpectations instance here when mutation is disabled to avoid surprises for the caller.

Reply:
Done